src/main/java/com/walmartlabs/x12/standard/AbstractTransactionSetParserChainable.java

```diff
@@ -0,0 +1,72 @@
+package com.walmartlabs.x12.standard;
+
+import com.walmartlabs.x12.X12Segment;
+import com.walmartlabs.x12.X12TransactionSet;
+
+import java.util.List;
+
+public abstract class AbstractTransactionSetParserChainable implements TransactionSetParser {
+
+    TransactionSetParser nextParser;
+    
+    /**
+     * chainable implementation of the {@link TransactionSetParser} interface
+     * 
+     * if this implementation does not handle the transaction set it will 
+     * pass it on to the next parser in the chain
+     */
+    @Override
+    public X12TransactionSet parseTransactionSet(List<X12Segment> transactionSegments, X12Group x12Group) {
+        if (handlesTransactionSet(transactionSegments, x12Group)) {
+            // we handle this transaction
+            // so parse it
+            return this.doParse(transactionSegments, x12Group);
+        } else if (nextParser != null) {
+            // we don't handle the transaction
+            // try the next link in the chain
+            return nextParser.parseTransactionSet(transactionSegments, x12Group);
+        } else {
+            // if we got here
+            // this transaction type is not handled
+            return null;
+        }
+    }
+    
+    /**
+     * convenience method that will allow one or more {@link TransactionSetParser} 
+     * to be registered w/ the parser
+     * if a transaction set type does not have a registered parser it is ignored
+     * @param transactionParser
+     */
+    public boolean registerNextTransactionSetParser(TransactionSetParser txParser) {
+        boolean isAdded = false;
```

Comment:
Should we verify that `txParser` isn't null? If it is null, should we return that it wasn't added even though I guess technically you could say that it was added

Reply:
well, unintended feature, this could allow consumer to clear out the currently registered chain of transaction set parsers. for now, will allow the "flexibility" to set it to null

Reply:
my bad. this would not clear out the chain... 
hmmm....